tests/run/rescue-boundary.scala

```diff
@@ -0,0 +1,76 @@
+import scala.util.control.{ControlException, NonFatal}
+import scala.util.*
+
+object lib:
+  extension [T](op: => T) inline def rescue (fallback: => T) =
+    try op
+    catch
+      case ex: ControlException => throw ex
+      case NonFatal(_) => fallback
+
+  extension [T, E <: Throwable](op: => T) inline def rescue (fallback: PartialFunction[E, T]) =
+    try op
+    catch
+      case ex: E =>
+        // user should never match `ReturnThrowable`, which breaks semantics of non-local return
```

Comment:
`ReturnThrowable`?

Reply:
That's just ported from the original `rescue` test. Not sure it made much sense there.